src/lib/components/events/event-carousel.svelte

```diff
@@ -24,11 +26,18 @@
 		event.preventDefault();
 		scrollTheCarousel(-event.deltaY);
 	};
+	onMount(() => {
+		hasHorizontalScrollbar = carouselRef.scrollWidth > carouselRef.clientWidth;
+	});
 </script>
 
 <section>
 	<div class="event-carousel-container">
-		<div bind:this={carouselButtonLeft} class="carousel-button left" on:click={scrollLeft}>
+		<div
+			bind:this={carouselButtonLeft}
+			class:enabled={hasHorizontalScrollbar}
```

Comment:
A separate variable `leftButtonEnabled` (and `rightButtonEnabled`) can be initialized and set to false adjacent to `hasHorizontalScrollbar`. Each time `scrollTheCarousel` is called, we should run:
```ts
leftButtonEnabled = carouselRef.scrollLeft > 0;
rightButtonEnabled = carouselRef.scrollWidth - carouselRef.scrollLeft - carouselRef.clientWidth < 1; // https://stackoverflow.com/a/42860948
```

Reply:
I added the leftButtonEnabled and rightButtonEnabled variables in the scrollTheCarousel function so it recalculates each time it's called. The only issue I had was that the first two initial clicks for the right button, scrollLeft is still zero then it begins to increment. So the left button appears after two clicks.